src/logic/map_objects/tribes/production_program.h

```diff
@@ -561,6 +561,15 @@ struct ProductionProgram : public MapObjectProgram {
 	const ProductionProgram::Groups& consumed_wares_workers() const;
 	const Buildcost& produced_wares() const;
 	const Buildcost& recruited_workers() const;
+	const std::string& trained_attribute() const {
+		return trained_attribute_;
+	}
+	const uint8_t& train_from_level() const {
+		return train_from_level_;
+	}
+	const uint8_t& train_to_level() const {
+		return train_to_level_;
+	}
```

Comment:
```suggestion
	uint8_t train_from_level() const {
		return train_from_level_;
	}
	uint8_t train_to_level() const {
		return train_to_level_;
	}
```
For primitive types, returning a copy is preferred over returning a const reference

Reply:
did not know about such finesses. However a String is not deemed a primitive in this case correct?